target/linux/ath79/dts/qca9558_dlink_dap-2680-a1.dts

```diff
@@ -0,0 +1,89 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+
+#include "qca955x.dtsi"
+#include "qca95xx_dlink_dap-2xxx.dtsi"
+
+/ {
+	compatible = "dlink,dap-2680-a1", "qca,qca9558";
+	model = "D-Link DAP-2680 A1";
+
+	aliases {
+		led-boot = &led_power_green;
+		led-failsafe = &led_power_red;
+		led-running = &led_power_green;
+		led-upgrade = &led_power_red;
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		reset {
+			label = "reset";
+			linux,code = <KEY_RESTART>;
+			gpios = <&gpio 18 GPIO_ACTIVE_LOW>;
+		};
```

Comment:
this usually has debounce-interval property set to 60...but I don't know if we are still doing this

Reply:
saw this with other devices, but the reset button also works quite well without...
I think it comes from the era of `gpio-keys-polled`, so I'm also not sure whether it is still a good idea when using interrupt-based GPIO.

Reply:
It is an important value, however, according to docs it has a default of 5, which may be good enough

the purpose is to avoid the electrical phenomena called "switch bounce" to be translated to 0 and 1 logic going back and forth during a single button press (in other words, 1 button press being perceived as many by cpu interrupt)

here is a good article if you're interested
https://www.eejournal.com/article/ultimate-guide-to-switch-debounce-part-2/

if you have an oscilloscope you're welcome to explore the best value, otherwise just leave it
it's possible that 60 was chosen as default without measuring an actual switch connection
some say it should be at least 20 ms...
but if it works it works

Reply:
That's why I've been wondering why (of all buttons) it is actually the _reset_ that some devices set a higher debounce value for - since it's mostly used to enter failsafe during boot (which does not provide immediate feedback by means of increasing the LED flashing frequency - this only happens after the failsafe time window - thus I typically end up pressing the button multiple times to make sure the device will successfully detect it).

When pressing reset during runtime, it needs to be held down for several seconds to force firstboot, otherwise it will just perform a simple reboot. Erroneously detecting a reboot request mutliple times within a few milliseconds should probably not make a difference, when the device would be going down for reboot anyways 🤔

In contrast, if it were e.g. a WiFi toggle switch, I agree it could be quite annoying when you are trying to enable wifi, but it would be disabled right away due to a bouncy button.

Reply:
60 ms is still a very small window. It takes more time to release and repress a button, unless you know someone who can press a button 16 times in 1 second 😄 

Yes, I believe the only concern is that a reboot will be faster than the factory reset trigger...I think it is possible depending on the device...

it may also be possible that the switch circuit for this board has some components that eliminate the bounce...

for failsafe, one only needs to make sure they don't press only once and too early...
immediate feedback would be nice, but the blink is already pretty fast...

Reply:
Just tried with the reset pin of DAP-3320, it takes several attempts to make it bounce upon being released, but I can barely make that phase longer than ~400µs, so if 5 ms is the default (is this stated somewhere in the driver source?) I guess that should be fine. There is one resistor in series and one pull-up, not sure how much these influence the actual behaviour though.

Reply:
yeah heres the docs btw

https://www.kernel.org/doc/Documentation/devicetree/bindings/input/gpio-keys.txt